### PR TITLE
Add sample for using blob storage like a filesystem

### PR DESCRIPTION
**Pull request by bsiegel**

### CODE REVIEW

**Comment by adxsdk6** [comment]
Can one of the admins verify this patch?

File: sdk/storage/azure-storage-blob/samples/directory_interface.py [hunk 1]
```diff
@@ -0,0 +1,303 @@
...
+    Fill in the required constant values and unzip directory_interface_data.zip
+    to run the samples below.
+'''
+
+import os
+from azure.storage.blob import BlobServiceClient
+
+# TODO: Fill in your connection string
+CONNECTION_STRING = ''
+CONTAINER_NAME = ''
```
Comment (tjprescott) [inline sdk/storage/azure-storage-blob/samples/directory_interface.py]: I'd recommend replacing this with code that pulls from environment variables:
```Python
import os
import sys

try:
  CONNECTION_STRING = os.environ['AZURE_STORAGE_CONNECTION_STRING']
except KeyError:
  print('AZURE_STORAGE_CONNECTION_STRING must be set.')
  sys.exit(1)
```

This way we don't have to worry about accidentally committing a credential and the script can theoretically be run "as-is".

Comment (tjprescott) [reply sdk/storage/azure-storage-blob/samples/directory_interface.py]: Container name could simply be an argument to the script.

Comment (bsiegel) [reply sdk/storage/azure-storage-blob/samples/directory_interface.py]: Good thought, it's addressed :)

**Comment by weshaggard** [comment]
Did you intend to include the zip file as part of this change?

**Comment by bsiegel** [comment]
> Did you intend to include the zip file as part of this change?

Yes, it lays down the example file/directory structure that is used by the sample. Kind of hate committing zips to the repo, so I am very much open to other suggestions.

**Comment by rakshith91** [comment]
Hi @bsiegel 
Thanks a lot.
Could you change the base branch to https://github.com/Azure/azure-sdk-for-python/tree/storage-samples-readme

**Comment by bsiegel** [comment]
Done. Also since it sounds like we want these samples to be runnable from PyTest, this zip file thing obviously won't work. I can just lay down the file structure in a setup method before the tests run, and delete them.

**Pull request merged**